pkg/readiness/ready_tracker.go

```diff
@@ -192,11 +193,126 @@ func (t *Tracker) Run(ctx context.Context) error {
 		return nil
 	})
 
+	// start deleted object polling. Periodically collects
+	// objects that are expected by the Tracker, but are deleted
+	ticker := time.NewTicker(500 * time.Millisecond)
+	grp.Go(func() error {
+		// wait 'holdoff' ticks before proceeding, hoping
+		// that the tracker will be satisfied by then
+		holdoff := 4
+		for i := 0; i < holdoff; _, i = <-ticker.C, i+1 {
+		}
+		for ; true; <-ticker.C {
+			if t.Satisfied(ctx) {
+				log.Info("readiness satisfied, no further collection")
+				ticker.Stop()
+				return nil
+			}
+			t.collectInvalidExpectations(ctx)
+		}
+
+		// effectively dead code to satisfy return type
+		return nil
+	})
+
 	_ = grp.Wait()
 	_ = t.constraintTrackers.Wait() // Must appear after grp.Wait() - allows trackConstraintTemplates() time to schedule its sub-tasks.
 	return nil
 }
 
+// collectForObjectTracker identifies objects that are unsatisfied for the provided
+// `es`, which must be an objectTracker, and removes those expectations
+func (t *Tracker) collectForObjectTracker(ctx context.Context, es Expectations) error {
+	if es == nil {
+		return fmt.Errorf("nil Expectations provided to collectForObjectTracker")
+	} else if !es.Populated() || es.Satisfied() {
+		log.Info("Expectations unpopulated or already satisfied, skipping collection")
+		return nil
+	}
+
+	// es must be an objectTracker so we can fetch `unsatisfied` expectations and get GVK
+	ot, ok := es.(*objectTracker)
+	if !ok {
+		return fmt.Errorf("Expectations was not an objectTracker in collectForObjectTracker")
```

Comment:
nit: lowercase error message. ref: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Note - I do believe this can happen in practice if a tracker is cancelled between .Keys() and .Get().

Reply:
`Get` keeps track of removed keys and avoid establishing new trackers for them afterwards, so in this case `es` would be an instance of `noopExpectations`.